local-cli/runAndroid/adb.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+var child_process = require('child_process');
+
+/**
+ * Parses the output of the 'adb devices' command
+ */
+function parseDevicesResult(result) {
+  if(!result) return;
+  
+  var devices = [];
+  var lines = result.trim().split(/\r?\n/);
+ 
+  for(var i=0; i<lines.length; i++) {
+    var words = lines[i].split(/[ ,\t]+/).filter(Boolean);
```

Comment:
I had to wrap my head around this one a bit, I guess it is to remove empty strings from the array?
If it is the case just write it as `.filter((w) => w !== '')` it is easier to understand.